### ISSUE TITLE
Supporting `HTTP_PROXY` / `http_proxy` environment variables globally

### ISSUE THREAD

**Issue by erain** [fluent/fluent-bit#2565]
**Is your feature request related to a problem? Please describe.**
It is a comment pattern for agents to support `HTTP_PROXY` or `http_proxy` environment variables for specifying a HTTP proxy, especially in containers ([reference](https://docs.docker.com/network/proxy/#use-environment-variables)) 

Particularly, fluentd's stackdriver output plugin support that ([document](https://cloud.google.com/logging/docs/agent/installation#configuring_an_http_proxy)) and we want to bring the feature parity in fluent-bit. 

**Describe the solution you'd like**

Provide `HTTP_PROXY` environment variable support in fluent-bit core (instead of in the output plugin).

**Describe alternatives you've considered**

Implement the `HTTP_PROXY` environment variable support in stackdriver output plugin, following a similar pattern as in the http output plugin ([src](https://github.com/fluent/fluent-bit/blob/1fb1e259da13f46d2d099f56a9f6cd2f4c49d6d3/plugins/out_http/http.c#L110-L114)).

**Comment by liuyuan10** [comment]
http output plugin is taking a different route by specifying proxy in the config instead of the HTTP_PROXY env variable. A question to the fluentbit, what is the right way here. My feeling is HTTP_PROXY should be supported by the underlying http library that applies to all plugins. Before that, individual plugin should use proxy option in the config just like the http output plugin?

**Comment by erain** [comment]
Thanks @liuyuan10 and I think that is a good idea. And that can be done by adding the `HTTP_PROXY` support directly in [flb_http_client](https://github.com/fluent/fluent-bit/blob/1fb1e259da13f46d2d099f56a9f6cd2f4c49d6d3/include/fluent-bit/flb_http_client.h#L137)

I will update the desired/alternative solutions and the title.

**Comment by erain** [comment]
@edsiper do you have any concern on adding support for `HTTP_PROXY` globally in fluent-bit? If not I will start the implementation.

Thank you!

**Comment by edsiper** [comment]
no concern! happy to receive the contribution.

- we have partial proxy support in http_client, please use out_http for testing
- when using environment vars HTTP_PROXY, HTTPS_PROXY, the core must be preset the values

**Comment by erain** [comment]
I have been working on this:
- https://github.com/erain/fluent-bit/commit/7afff9688e48c7917b9401ba90d49162d2e62e42
- https://github.com/erain/fluent-bit/commit/3c6d8475079cae24d6d7d5ddd8a5acb30e33a870

Things worked well with HTTP proxy + HTTP output. 

However, when I tested HTTP proxy + Stackdriver output (HTTPS) things stopped working. And it looks like I messed up in the `flb_upstream` and I am kinda stuck. 

@edsiper : any suggestions / tips for creating a `flb_upstream` when a proxy is using?
